src/anyio/_core/_tasks.py

```diff
@@ -156,3 +180,248 @@ def create_task_group() -> TaskGroup:
 
     """
     return get_async_backend().create_task_group()
+
+
+class TaskHandle(Generic[T]):
```

Comment:
It looks like a `TaskHandle` could be used to implement a `create_future()` on the `EnhancedTaskGroup`, to have something like asyncio's `Future`.
What do you think?

Reply:
What exact addition are you suggesting?

Reply:
I was going to say something like that:
```py
class EnhancedTaskGroup:
    def create_future(self) -> TaskHandle[T]:
        handle = TaskHandle[T]()
        return handle
```
But now I realize a `Future` is just a `TaskHandle`, so there's nothing to do?